images_to_samples.py

```diff
@@ -450,22 +452,14 @@ def main(params):
                     raise ValueError(f"Dataset value must be trn or tst. Provided value is {info['dataset']}")
                 val_file = val_hdf5
 
-                # TODO: save histogram to metadata, then use it at radiometric trimming, if chosen.
-                metadata = raster.meta
-                metadata['name'] = raster.name
-                metadata['csv_info'] = info
+                metadata = add_metadata_from_raster_to_sample(sat_img_arr=np_input_image,
```

Comment:
Can we wrap metadata functionality in a condition? This would resolve some errors that may arise when metadata for a raster does not exist or if it is incorrect.

Reply:
There are occasions where the raster's metadata are empty. In my case it was an error with my Gdal installation and Rasterio wasn't throwing any errors with empty CRS in the raster.   ([fix for GDAL here](https://stackoverflow.com/questions/56764046/gdal-ogr2ogr-cannot-find-proj-db-error#:~:text=3%20Answers&text=You%20might%20need%20to%20set%20the%20PROJ_LIB%20environment%20variable.&text=If%20you%20don't%20find,see%20if%20things%20start%20working))

**Question is :** Are there negative impacts in our process if we have incorrect metadata?   
- If so, we should handle such case.   
- If we don't do much with the metadata (it's just info about the original raster), a simple warning could do.

Reply:
the metadata is only used for 2 things during training: 
1. knowing what's the original datatype for scaling (ex.: if scaling knows original range is 0-255, it will then scale all values from that to 0-1. An exception handling behavior if metadata is missing would be to scan the min and max values in array and deduce a datatype based on that. Would that be acceptable ?
2. random radiometric trim: the histogram of source image is used to determine which value is at 2nd percentile, for example. It keeps GDL from computing histogram from cropped sample, which is not representative enough and is also longer to compute. There can't be any problem with this step though, since the histogram can always be computed at sample prep time: it's just a bincount on the source image saved to a dictionary.

Reply:
I don't think it's required at this time. Maybe we can file an issue about it and we'll discuss the best way to go from there.  
GDAL will raise a warnings for the proj.db file missing, related to the issue I got.

Reply:
I would run some tests on this.